spec/controllers/post_controller_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+    describe "#new" do
+        it "responds successfully" do
+            post_params = FactoryBot.attributes_for(:post)
+            post :new, params: { post: post_params }
+            expect(response).to be_successful
+        end
+
+        it "returns a 200" do
+            post_params = FactoryBot.attributes_for(:post)
+            post :new, params: { post: post_params }
+            expect(response).to have_http_status "200"
+        end
+    end
+
+    describe "#create" do
+        context "with valid attributes" do
+            it "returns a 302" do
+                post_params = FactoryBot.attributes_for(:post)
+                post :create, params: { post: post_params }
+                expect(response).to have_http_status "302"
+            end
+
+            it "redirects correctly" do
+                @post = FactoryBot.create(:post)
```

Comment:
This looks like it will create two posts as part of this spec

Reply:
Yup, makes 2 and you're right I was having issues getting the ID from the post I just created. For some reason, though the ID I get from the path is continuing to go up. So if I run it now it's in the 60's. That's why I think something is up with the test database as it doesn't seem to reset each time.